File: 0735-asteroid-collision/0735-asteroid-collision.cpp
class Solution {
public:
    vector<int> asteroidCollision(vector<int>& arr) {
        stack<int>s;
        int n = arr.size();

        for(int i = 0 ;i< n ; i++){

            if(arr[i]>0)s.push(arr[i]);
            else{
            while(!s.empty() && s.top()>0 && s.top()<abs(arr[i]))s.pop();
            if(!s.empty() && s.top()==abs(arr[i]))s.pop();
            else if(s.empty() || s.top()<0)s.push(arr[i]);
            }
        }
        vector<int>ans;
        while(!s.empty()){
            ans.push_back(s.top());
            s.pop();
        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: 0735-asteroid-collision/0735-asteroid-collision.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        
        st=deque()

        for a in asteroids:
            if a>0:
                st.append(a)
            else:
                while st and st[-1]>0 and st[-1]<abs(a):
                    st.pop()
                if(st and st[-1]==abs(a)):
                    st.pop()
                elif not st or st[-1]<0:
                    st.append(a)
        return list(st)

                